osu.Game/Input/Bindings/RealmKeyBinding.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Input.Bindings;
+using osu.Game.Database;
+using Realms;
+
+namespace osu.Game.Input.Bindings
+{
+    [MapTo(nameof(KeyBinding))]
+    public class RealmKeyBinding : RealmObject, IHasGuidPrimaryKey, IKeyBinding
+    {
+        [PrimaryKey]
+        public string StringGuid { get; set; }
+
+        [Ignored]
+        public Guid ID
+        {
+            get => Guid.Parse(StringGuid);
+            set => StringGuid = value.ToString();
+        }
+
+        public int? RulesetID { get; set; }
+
+        public int? Variant { get; set; }
+
+        public KeyCombination KeyCombination
+        {
+            get => KeyCombinationString;
+            set => KeyCombinationString = value.ToString();
+        }
+
+        public object Action
+        {
+            get => ActionInt;
+            set => ActionInt = (int)value;
+        }
+
+        [MapTo(nameof(Action))]
+        public int ActionInt { get; set; }
```

Comment:
So I assume this means that internally this is weaved into a property that does `set => Action = (object)value`?

Reply:
no, the mapping is just for naming purposes.

see realm docs: https://docs.mongodb.com/realm-legacy/docs/dotnet/latest/